api/locationSlice.ts

```diff
@@ -0,0 +1,24 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from '@reduxjs/toolkit';
+
+export interface locationState {
+  locationInfo: string
+}
+
+const initialState: locationState = {
+  locationInfo: "",
+}
+
+
+const locationSlice = createSlice({
+  name: "location",
+  initialState,
+  reducers: {
+    getLocation: (state, action: PayloadAction<string>) => {
+      state.locationInfo = action.payload;
+    },
```

Comment:
We should rename this one as well, as it's performing a set rather than a get.  Let's change this to `setLocation`.